android/quest/src/main/java/org/smartregister/fhircore/quest/ui/report/measure/MeasureReportViewModel.kt

```diff
@@ -282,24 +282,36 @@ constructor(
             // Show Progress indicator while evaluating measure
             toggleProgressIndicatorVisibility(true)
             val result =
-              measureReportConfigList.flatMap {
-                retrievePreviouslyGeneratedMeasureReports<MeasureReport>(
-                  fhirEngine,
-                  startDateFormatted,
-                  endDateFormatted,
-                  it.url
-                )
-                  .ifEmpty {
-                    withContext(dispatcherProvider.io()) {
-                      fhirEngine.loadCqlLibraryBundle(fhirOperator, it.url)
-                    }
-                    evaluatePopulationMeasure(
-                      it.url,
-                      startDateFormatted,
-                      endDateFormatted,
-                      it.subjectXFhirQuery
-                    )
+              measureReportConfigList.flatMap { config ->
+                val existing =
+                  retrievePreviouslyGeneratedMeasureReports<MeasureReport>(
+                    fhirEngine,
+                    startDateFormatted,
+                    endDateFormatted,
+                    config.url
+                  )
+
+                // if report is of current month or does not exist generate a new one and replace
+                // existing
+                if (startDateFormatted.contentEquals(
```

Comment:
Don't the start and the end date need to match?

Reply:
the start and end dates in reports are start or end of months.. so if start date is of current month or end date is falling in current month we need to regenerate the reports.

Reply:
OK got it, I had the condition reversed, but something still feels odd here. One thing to consider is that reports don't have to only be for a month.

If a report starts on the 5th of the month, and ends on the 15th of the month, we'd want to use the existing report as long as end date is in the past. (I'm assuming there's some logic elsewhere requiring that end date is after start date, can you check date please?)

Maybe this check is simpler, if the end date is before the current date and there is an existing report, load it, otherwise generate.

Can you add some tests around reports that are less than and greater than 1 month?